src/stream.rs

```diff
@@ -163,8 +163,33 @@ where
     }
 
     /// Sets preferred output buffer size.
-    /// Larger buffer size causes longer playback delays. Buffer sizes that are too small
-    /// may cause higher CPU usage or playback interruptions.
+    ///
+    /// To play sound without any glitches the audio card may never receive a
+    /// sample to late. Some samples might take longer to generate then
+    /// others. For example because:
+    ///  - The OS preempts the thread creating the samples. This happens more
+    ///    often if the computer is under high load.
+    ///  - The decoder needs to read more data from disk.
+    ///  - Rodio code takes longer to run for some samples then others
+    ///  - The OS can only send audio samples in groups to the DAC.
+    ///
+    /// The OS solves this by buffering samples. The larger that buffer the
+    /// smaller the impact of variable sample generation time. On the other
+    /// hand Rodio controls audio by changing the value of samples. We can not
+    /// change a sample already in the OS buffer. That means there is a
+    /// minimum delay (latency) of `<buffer size>/<sample_rate*channel_count>`
+    /// seconds before a change made through rodio takes effect.
+    ///
+    /// ## Large vs Small buffer
+    /// - A larger buffer size results in high latency. Changes made trough
+    ///   Rodio (volume/skip/effects etc) takes longer before they can be heard.
+    /// - A small buffer might cause higher CPU usage, playback interruptions
+    ///   or result in errors like: `alsa::poll() returned POLLERR`
+    ///
+    /// ## Recommendation
```

Comment:
Would we want to mention some typical values, like:
- low-latency (audio production, live monitoring): 512-1024
- general use (games, media playback): 1024-2048
- stability-focused (background music, non-interactive): 2048-4096

A reason *not* to do it, is that this is somewhat opinionated.
Then again, it could put novices on the right track.